src/Puzzle.java

```diff
@@ -1,79 +1,98 @@
+import java.util.*;
 
 public class Puzzle {
 	private Box[][] puzzle;
 
-	public Puzzle() {
-		this.puzzle = buildPuzzle();
-	}
+	public Puzzle(String[] nums) {
+		this.puzzle = new Box[9][9];
 
-	public Box[][] buildPuzzle() {
-		Box[][] puzzle = new Box[9][9];
 		for (int i = 0; i < puzzle.length; i++) {
 			for (int j = 0; j < puzzle[i].length; j++) {
-				Box box = new Box();
-				puzzle[i][j] = box;
+				if (Character.isDigit(nums[i].charAt(j))) {
+					puzzle[i][j] = new Box(Character.getNumericValue(nums[i].charAt(j)));
+				} else {
+					puzzle[i][j] = new Box();
+				}
 			}
 		}
-		return puzzle;
-	}
-
-	public void setValue(int row, int col, int value) {
-		this.puzzle[row][col].setValue(value);
 	}
 
-	public void printPuzzle() {
+	private void printChoices() {
 		for (int i = 0; i < puzzle.length; i++) {
 			for (int j = 0; j < puzzle[i].length; j++) {
-				int val = this.puzzle[i][j].getValue();
-				if (val == 0) {
-					System.out.print(" ");
-				} else {
-					System.out.print(val);
+				if (!this.puzzle[i][j].isSolved()) {
+					System.out.println("Choices left " + this.puzzle[i][j].numChoices());
 				}
 			}
-			System.out.println();
 		}
 	}
 
-	public void solvePuzzle() {
-		boolean done =false; 
-		int count = 0;
-		while (!done) {
-			count++; 
-			System.out.println("iteration " + count);
-			done=true; 
-			for (int i = 0; i < puzzle.length; i++) {
-				for (int j = 0; j < puzzle[i].length; j++) {
-					Box box = this.puzzle[i][j];
-					if (!box.isSolved()) {
-						System.out.println("box" + i + j + "is not finished");
-						done=false; 
-						checkRow(i, box.getValue());
-						checkCol(j, box.getValue());
-						
-					}
+	private void removeKnownValues() {
```

Comment:
This finds all values in the current row and removes those from the possible values for the current cell.  The same goes for the current cell's row and it square.